middleware.js

```diff
@@ -52,6 +52,20 @@ function pickLanguage (req, res, next) {
 }
 
 
+async function recordVisitFromEmail (req, res, next) {
+  if (req.query.utm_source && req.query.utm_source !== "fx-monitor-email") {
+    next();
+  }
+  if (!req.query.subscriber_id || !Number.isInteger(req.query.subscriber_id)) {
+    next();
+  }
+  const subscriber = await DB.getSubscriberById(req.query.subscriber_id);
+  const fxaMetricsFlowPath = `metrics-flow?event_type=engage&uid=${subscriber.fxa_uid}&service=${AppConstants.OAUTH_CLIENT_ID}`;
```

Comment:
What are the chances there that `DB.getSubscriberById()` returns nothing, `subscriber` is null/undefined, and then `${subscriber.fxa_uid}` throws up?

Reply:
High enough to check for the property first!

Reply:
Updated.